fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -306,4 +306,21 @@ public void noPrimaryKey() {
                 RecordMetaData.newBuilder().setRecords(TestRecordsNoPrimaryKeyProto.getDescriptor()).getRecordMetaData());
         assertEquals("Record type MyNoPrimaryKeyRecord must have a primary key", t.getMessage());
     }
+
+    @Test
+    public void updateRecords() {
+        RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
+        assertNotNull(builder.getRecordType("MySimpleRecord"));
+        assertTrue(builder.getRecordType("MySimpleRecord").getDescriptor().getFile() == TestRecords1Proto.getDescriptor());
+        assertNotNull(builder.getRecordType("MyOtherRecord"));
+        assertTrue(builder.getRecordType("MyOtherRecord").getDescriptor().getFile() == TestRecords1Proto.getDescriptor());
+        builder.updateRecords(TestRecords1EvolvedProto.getDescriptor(), true);
+        RecordMetaData recordMetaData = builder.build(true);
+        assertNotNull(recordMetaData.getRecordType("MySimpleRecord"));
+        assertTrue(builder.getRecordType("MySimpleRecord").getDescriptor().getFile() == TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("MyOtherRecord"));
+        assertTrue(builder.getRecordType("MyOtherRecord").getDescriptor().getFile() == TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("AnotherRecord"));
+        assertTrue(builder.getRecordType("AnotherRecord").getDescriptor().getFile() == TestRecords1EvolvedProto.getDescriptor());
+    }
```

Comment:
Probably worth validating that calling `updateRecords` when `local_file_descriptor` is already set results in an error.